modules/vertamediaBidAdapter.js

```diff
@@ -1,119 +1,125 @@
-      w: size.width,
-      h: size.height,
-      domain: document.location.hostname
-    };
-  }
-
-  function getSize(requestSizes) {
-    const parsed = {};
-    const size = utils.parseSizesInput(requestSizes)[0];
-
-    if (typeof size !== 'string') {
-      return parsed;
-    }
-
-    let parsedSize = size.toUpperCase().split('X');
-
-    return {
-      width: parseInt(parsedSize[0], 10) || undefined,
-      height: parseInt(parsedSize[1], 10) || undefined
-    };
-  }
-
-  /* Notify Prebid of bid responses so bids can get in the auction */
-  function handleResponse(response) {
-    var parsed;
-
-    try {
-      parsed = JSON.parse(response);
-    } catch (error) {
-      utils.logError(error);
-    }
-
-    if (!parsed || parsed.error || !parsed.bids || !parsed.bids.length) {
-      bidmanager.addBidResponse(bidRequest.placementCode, createBid(STATUS.NO_BID));
+    serverResponse.bids.forEach(serverBid => {
+      if (serverBid.cpm !== 0) {
+        const bid = createBid(serverBid, bidderRequest);
+        bids.push(bid);
+      }
+    });
 
-      return;
-    }
+    return bids;
+  },
+};
+
+/**
+ * Prepare all parameters for request
+ * @param bid {object}
+ * @returns {object}
+ */
+function prepareRTBRequestParams(bid) {
+  let size = getSize(bid.sizes);
+
+  return {
+    domain: utils.getTopWindowLocation().hostname,
+    callbackId: bid.bidId,
+    aid: bid.params.aid,
+    h: size.height,
+    w: size.width
+  };
+}
 
-    bidmanager.addBidResponse(bidRequest.placementCode, createBid(STATUS.GOOD, parsed.bids[0]));
+/**
+ * Prepare size for request
+ * @param requestSizes {array}
+ * @returns {object} bid The bid to validate
+ */
+function getSize(requestSizes) {
+  const size = utils.parseSizesInput(requestSizes)[0];
+  const parsed = {};
+
+  if (typeof size !== 'string') {
+    return parsed;
   }
 
-  function createBid(status, tag) {
-    var bid = bidfactory.createBid(status, tag);
+  let parsedSize = size.toUpperCase().split('X');
 
-    bid.code = baseAdapter.getBidderCode();
-    bid.bidderCode = bidRequest.bidder;
-
-    if (!tag || status !== STATUS.GOOD) {
-      return bid;
-    }
-
-    bid.mediaType = 'video';
-    bid.cpm = tag.cpm;
-    bid.creative_id = tag.cmpId;
-    bid.width = bidRequest.width;
-    bid.height = bidRequest.height;
-    bid.descriptionUrl = tag.url;
-    bid.vastUrl = tag.url;
-
-    return bid;
-  }
-
-  return Object.assign(this, {
-    callBids: baseAdapter.callBids,
-    setBidderCode: baseAdapter.setBidderCode
-  });
+  return {
+    height: parseInt(parsedSize[1], 10) || undefined,
+    width: parseInt(parsedSize[0], 10) || undefined
+  };
 }
 
-adaptermanager.registerBidAdapter(new VertamediaAdapter(), 'vertamedia', {
-  supportedMediaTypes: ['video']
-});
+/**
+ * Configure new bid by response
+ * @param bidRequest {object}
+ * @param bidResponse {object}
+ * @returns {object}
+ */
+function createBid(bidResponse, bidRequest) {
+  return {
+    bidderCode: bidRequest.bidderCode,
```

Comment:
`bidderCode` will be set by bidderFactory. Can be removed from here.
Also no need to add `descriptionUrl`